File: src/Helpers/ProfileFileHelpers.cs
using System;
using System.IO;

public static class ProfileFileHelpers
{
    /// <summary>
    /// Loads a profile by name into the configuration store.
    /// </summary>
    /// <param name="profileName">The name of the profile to load</param>
    /// <exception cref="CommandLineException">Thrown when profile name is empty or profile not found</exception>
    public static void LoadProfile(string profileName)
    {
        if (string.IsNullOrEmpty(profileName))
        {
            throw new CommandLineException("Profile name cannot be empty.");
        }
        
        var profilePath = FindProfileFile(profileName);
        if (profilePath == null)
        {
            throw new CommandLineException($"Profile '{profileName}' not found in any scope or parent directories.");
        }
        
        ConsoleHelpers.WriteDebugLine($"Loading profile from {profilePath}");
        ConfigStore.Instance.LoadConfigFile(profilePath);
    }
    
    /// <summary>
    /// Finds a profile file by name across all scopes and optionally parent directories.
    /// </summary>
    /// <param name="profileName">The name of the profile to find</param>
    /// <returns>The path to the profile file if found, null otherwise</returns>
    public static string? FindProfileFile(string profileName)
    {
        var yamlFileName = $"{profileName}.yaml";
        var yamlProfilePath = ScopeFileHelpers.FindFileInAnyScope(yamlFileName, "profiles", searchParents: true);
        if (yamlProfilePath != null)
        {
            return yamlProfilePath;
        }

        var bareFileNameNoExt = $"{profileName}";
        return ScopeFileHelpers.FindFileInAnyScope(bareFileNameNoExt, "profiles", searchParents: true);
    }
    
    /// <summary>
    /// Finds the profiles directory in the specified scope.
    /// </summary>
    /// <param name="scope">The scope to search in</param>
    /// <param name="create">Whether to create the directory if it doesn't exist</param>
    /// <returns>The path to the profiles directory, or null if not found</returns>
    public static string? FindProfilesDirectoryInScope(ConfigFileScope scope, bool create = false)
    {
        return create
            ? ScopeFileHelpers.EnsureDirectoryInScope("profiles", scope)
            : ScopeFileHelpers.FindDirectoryInScope("profiles", scope);
    }
}

File: src/commonjs/Helpers/ProfileFileHelpers.ts
import { ConfigFileScope } from '../Configuration/ConfigFileScope';
import { ScopeFileHelpers } from '../Configuration/ScopeFileHelpers';
import { ConsoleHelpers } from './ConsoleHelpers';
import { ConfigStore } from '../Configuration/ConfigStore';

export class CommandLineException extends Error {
    constructor(message: string) {
        super(message);
        this.name = 'CommandLineException';
    }
}

export class ProfileFileHelpers {
    /**
     * Loads a profile by name into the configuration store.
     * @param profileName The name of the profile to load
     * @throws CommandLineException Thrown when profile name is empty or profile not found
     */
    public static loadProfile(profileName: string): void {
        if (!profileName || profileName.length === 0) {
            throw new CommandLineException('Profile name cannot be empty.');
        }
        
        const profilePath = ProfileFileHelpers.findProfileFile(profileName);
        if (profilePath === null) {
            throw new CommandLineException(`Profile '${profileName}' not found in any scope or parent directories.`);
        }
        
        ConsoleHelpers.writeDebugLine(`Loading profile from ${profilePath}`);
        ConfigStore.instance.loadConfigFile(profilePath);
    }
    
    /**
     * Finds a profile file by name across all scopes and optionally parent directories.
     * @param profileName The name of the profile to find
     * @returns The path to the profile file if found, null otherwise
     */
    public static findProfileFile(profileName: string): string | null {
        const yamlFileName = `${profileName}.yaml`;
        const yamlProfilePath = ScopeFileHelpers.findFileInAnyScope(yamlFileName, 'profiles', true);
        if (yamlProfilePath !== null) {
            return yamlProfilePath;
        }

        const bareFileNameNoExt = `${profileName}`;
        return ScopeFileHelpers.findFileInAnyScope(bareFileNameNoExt, 'profiles', true);
    }
    
    /**
     * Finds the profiles directory in the specified scope.
     * @param scope The scope to search in
     * @param create Whether to create the directory if it doesn't exist
     * @returns The path to the profiles directory, or null if not found
     */
    public static findProfilesDirectoryInScope(scope: ConfigFileScope, create: boolean = false): string | null {
        return create
            ? ScopeFileHelpers.ensureDirectoryInScope('profiles', scope)
            : ScopeFileHelpers.findDirectoryInScope('profiles', scope);
    }
}